lms/djangoapps/instructor_task/api.py

```diff
@@ -437,17 +437,19 @@ def submit_export_ora2_data(request, course_key):
     return submit_task(request, task_type, task_class, course_key, task_input, task_key)
 
 
-def generate_certificates_for_students(request, course_key, students=None):  # pylint: disable=invalid-name
+def generate_certificates_for_students(request, course_key, student_set=None, specific_student_id=None):  # pylint: disable=invalid-name
     """
-    Submits a task to generate certificates for given students enrolled in the course or
-    all students if argument 'students' is None
+    Submits a task to generate certificates for given students enrolled in the course.
 
     Raises AlreadyRunningError if certificates are currently being generated.
     """
-    if students:
-        task_type = 'generate_certificates_certain_student'
-        students = [student.id for student in students]
-        task_input = {'students': students}
+    if student_set:
+        task_type = 'generate_certificates_student_set'
+        task_input = {'student_set': student_set}
+
+        if student_set == 'specific_student':
+            task_type = 'generate_certificates_certain_student'
+            task_input.update({'specific_student_id': specific_student_id})
```

Comment:
@adampalay can you tell us the scenario in which `specific_student_id` is `None`.